codeamor/2020-12-19-typescript.md

```diff
@@ -0,0 +1,187 @@
+# **TypeScript 개념 및 기본 용법**
+
+```js
+let v1 = 123;
+v1 = 'abc';
+```
+
+TypeScript에서 위와 같이 선언 시 두 번째 v1은 타입 에러가 발생합니다.<br>
+이는 첫 번째 할당 시 문자열이 아닌 숫자가 변수에 담겼기 때문입니다.<br>
+이를 타입 추론(type inference)이라고 합니다.<br>
+두 가지 모두 변수에 담기 위해서는 두 가지 타입을 명시해 주어야 합니다.<br>
+
+```js
+let v1: number | string = 123;
+v1 = 'abc';
+```
+any 타입은 모든 종류의 값을 허용합니다.<br>
```

Comment:
[Typescript](https://www.typescriptlang.org/docs/handbook/basic-types.html#any) 문서를 보시면 any 사용을 피하라는 내용이 있는데요..! 그 내용이 포함되면 좋을거같아요.

> After all, remember that all the convenience of any comes at the cost of losing type safety. Type safety is one of the main motivations for using TypeScript and you should try to avoid using any when not necessary.